lib/modules/apostrophe-pieces/public/js/manager-modal.js

```diff
@@ -567,8 +565,28 @@ apos.define('apostrophe-pieces-manager-modal', {
         } else {
           self.removeChoice(id);
         }
+
+        var gridIds = self.$el.find('[data-piece]').map(function() {
+          return $(this).attr('data-piece');
+        }).get();
+
+        if (self.choices.length === gridIds.length) {
```

Comment:
Choices might not have been made on the same page. You need to make sure they are the same ids.

Reply:
This is only checking for a normal select piece checkbox that happens to complete all selections on a page .. reaching beyond the page still triggers UI changes in the usual way.

Reply:
Right, but you're testing that by checking to see if the number of choices matches the number of ids on the page, and they could've made those choices on page two and then navigated to page three (you can do that).